resources/assets/components/UserIndex.js

```diff
@@ -0,0 +1,109 @@
+import gql from 'graphql-tag';
+import { Link } from 'react-router-dom';
+import { useQuery } from '@apollo/react-hooks';
+import React, { useRef, useState } from 'react';
+
+import Empty from './Empty';
+import UserShell from './UserShell';
+
+const USER_SEARCH_QUERY = gql`
+  query UserSearchQuery($term: String!) {
+    users(search: $term) {
+      id
+      displayName
+      emailPreview
+      mobilePreview
+    }
+  }
+`;
+
+/**
+ * Load & display search results for the given term.
+ *
+ * @param {string} term
+ */
+const SearchResults = ({ term }) => {
+  const { loading, error, data } = useQuery(USER_SEARCH_QUERY, {
+    variables: { term },
+  });
+
+  if (loading) return <div className="spinner" />;
+  if (error) return <span>Error :(</span>;
+
+  if (data.users.length === 0) {
+    return <Empty />;
+  }
+
+  return (
+    <table className="table">
+      <thead>
+        <tr className="table__header">
+          <td className="table__cell">Name</td>
+          <td className="table__cell">Email</td>
+          <td className="table__cell">Mobile</td>
+        </tr>
+      </thead>
+      <tbody>
+        {data.users.map(user => (
+          <tr className="table__row">
+            <td className="table__cell">
+              <Link to={`/users/${user.id}`}>{user.displayName}</Link>
+            </td>
+            <td className="table__cell">{user.emailPreview}</td>
+            <td className="table__cell">{user.mobilePreview}</td>
+          </tr>
+        ))}
+      </tbody>
+    </table>
+  );
+};
+
+/**
+ * The user index!
+ *
+ * @param {string} term
+ */
+const UserIndex = () => {
+  const [term, setTerm] = useState(null);
+  const inputElement = useRef(null);
```

Comment:
Here's [documentation for the `useRef` hook](https://reactjs.org/docs/hooks-reference.html#useref) if you're interested!

Reply:
> Essentially, useRef is like a “box” that can hold a mutable value in its .current property.

a-ha! thanks for including the link 😸